crates/pathfinder/src/rpc.rs

```diff
@@ -74,342 +65,23 @@ Hint: If you are looking to run two instances of pathfinder, you must configure
-
-        let method_callback = move |params: Params<'static>, context: Arc<RpcContext>| {
-            // why info here? it's the same used in warp tracing filter for example.
-            let span = tracing::info_span!("rpc_method", name = method_name);
-            async move {
-                let input = params.parse::<Input>()?;
-                method((*context).clone(), input).await.map_err(|err| {
-                    let rpc_err: error::RpcError = err.into();
-                    jsonrpsee::core::Error::from(rpc_err)
-                })
-            }
-            .instrument(span)
-        };
+        let context_v02 = (&self.api).into();
 
-        module
-            .register_async_method(method_name, method_callback)
-            .with_context(|| format!("Registering {method_name}"))?;
-
-        Ok(())
-    }
-
-    /// Registers a JSON-RPC method with the [RpcModule<RpcContext>](jsonrpsee::RpcModule).
-    ///
-    /// An example signature for `method` is:
-    /// ```ignore
-    /// async fn method(context: RpcContext) -> Result<Ouput, Error>
-    /// ```
-    #[allow(dead_code)]
-    fn register_method_with_no_input<Output, Error, MethodFuture, Method>(
-        module: &mut jsonrpsee::RpcModule<RpcContext>,
-        method_name: &'static str,
-        method: Method,
-    ) -> anyhow::Result<()>
-    where
-        Output: 'static + ::serde::Serialize + Send + Sync,
-        Error: Into<error::RpcError>,
-        MethodFuture: std::future::Future<Output = Result<Output, Error>> + Send,
-        Method: (Fn(RpcContext) -> MethodFuture) + Copy + Send + Sync + 'static,
-    {
-        use anyhow::Context;
-        use std::sync::Arc;
-        use tracing::Instrument;
-
-        metrics::register_counter!("rpc_method_calls_total", "method" => method_name);
-
-        let method_callback = move |_params, context: Arc<RpcContext>| {
-            // why info here? it's the same used in warp tracing filter for example.
-            let span = tracing::info_span!("rpc_method", name = method_name);
-            async move {
-                method((*context).clone()).await.map_err(|err| {
-                    let rpc_err: error::RpcError = err.into();
-                    jsonrpsee::core::Error::from(rpc_err)
-                })
-            }
-            .instrument(span)
-        };
+        let mut module_v01 = v01::RpcModuleWrapper::new(RpcModule::new(self.api));
+        v01::register_all_methods(&mut module_v01)?;
+        let module_v01: Methods = module_v01.into_inner().into();
 
-        module
-            .register_async_method(method_name, method_callback)
-            .with_context(|| format!("Registering {method_name}"))?;
+        let mut module_v02 = RpcModule::new(context_v02);
+        v02::register_all_methods(&mut module_v02)?;
+        let module_v02 = module_v02.into();
 
-        Ok(())
+        Ok(server
+            .start_with_paths([
+                (vec!["/", "/rpc/v01"], module_v01),
```

Comment:
Is it possible or even correct to have `/rpc/v0.1`?

Reply:
Let's check :thinking:

Reply:
Dots are allowed in URL paths. :tada: